Library/Homebrew/rubocops/formula_desc_cop.rb

```diff
@@ -0,0 +1,78 @@
+require_relative "./extend/formula_cop"
+require_relative "../extend/string"
+
+module RuboCop
+  module Cop
+    module Homebrew
+      # This cop audits `desc` in Formulae
+      #
+      # - Checks for existence of `desc`
+      # - Checks if size of `desc` > 80
+      # - Checks if `desc` begins with an article
+      # - Checks for correct usage of `command-line` in `desc`
+      # - Checks if `desc` contains the formula name
+      class FormulaDesc < FormulaCop
+        def audit_formula(node, class_node, _parent_class_node, body)
```

Comment:
While trying to implement the above I faced an issue. Just by having 
```ruby
if desc =~ /(command ?line)/i
  problem "Description should use \"command-line\" instead of \"#{$1}\""
end
```
its difficult to set all the instance variables required for pinpointing the `source_range` of the style violation. So I added a method `check_regex_match`. Now the code looks like 
```ruby
match_object = check_regex_match(desc, /(command ?line)/i)
if match_object
  problem "Description should use \"command-line\" instead of \"#{match_object}\""
end
```